### PR TITLE
DOC: added plotting module to the api reference docs

### LINKED ISSUE #19777: API reference is missing the Plotting module
I noticed this when looking for the documentation page for the `scatter_matrix` function.

#### Problem description

I needed to check the parameters I needed to pass to `plotting.scatter_matrix`. Had to go all the way to the source code.

Turns out none of the functions in `_misc.py` are included in the API reference. I added a **Plotting** section after **General functions** and before **Series**. Under that section I added a **Miscellaneous** section which includes:

- scatter_matrix
- radviz
- andrews_curves
- bootstrap_plot
- parallel_coordinates
- lag_plot

### PR DESCRIPTION
**Pull request by mehemken**
- [x ] closes #19777
- [ ] tests added / passed
- [ ] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

**Review by jreback — CHANGES_REQUESTED** [review]
you seem to have added a bunch of files to the repo

**Comment by mehemken** [comment]
@jreback Thanks for the quick feedback. The only changes I made were to doc/source/api.rst. Those files were automatically generated by the docs. Should I not include them in the commit? They look like they might be part of the examples.

**Comment by jreback** [comment]
no your don’t need to include them

File: doc/source/api.rst [hunk 1]
```diff
@@ -246,6 +246,24 @@ Testing
...
    test
 
+.. currentmodule:: pandas.plotting
+
+Plotting
+--------
+
+Miscellaneous
+~~~~~~~~~~~~~
+
```
Comment (jorisvandenbossche) [inline doc/source/api.rst]: Can you add a small sentence saying that those functions are contained in the `pandas.plotting` module?

File: doc/source/api.rst [hunk 2]
```diff
@@ -246,6 +246,24 @@ Testing
 
    test
 
+.. currentmodule:: pandas.plotting
+
+Plotting
+--------
```
Comment (jorisvandenbossche) [inline doc/source/api.rst]: Can you move this section more to the bottom? There is already a small section on plotting that includes `(de)register_matplotlib_converters`, so you can merge both

**Review by jorisvandenbossche — COMMENTED** [review]
The actual change looks good, thanks! Added a few comments.

But to clarify @jreback comment: you only need to include those changes in api.rst itself, not all the other files. So you will need to remove them again from your branch (those files are generated by the doc building process, so don't need to be in the git repo)

File: doc/source/api.rst (hunk 1 above)
Comment (mehemken) [reply doc/source/api.rst]: I'll add this too.

File: doc/source/api.rst (hunk 2 above)
Comment (mehemken) [reply doc/source/api.rst]: @jorisvandenbossche Sure thing! Thanks for the feedback. Would it matter if I just did this in a new branch? I'm having trouble crafting the right git command that will un-commit and unstage my changes. Or is it preferred I keep everything on this branch? Or should I just put it in master?

Comment (jorisvandenbossche) [reply doc/source/api.rst]: Normally you should be able to do `git rm doc/source/savefig/*` and commit those changes. No need to un-commit the previous one (we will squash all commits into one in the end anyway).

Comment (jorisvandenbossche) [reply doc/source/api.rst]: So if that works, we would prefer that you further push to this branch / PR.

Comment (mehemken) [reply doc/source/api.rst]: Got it. I'll do that now.

**Comment by jorisvandenbossche** [comment]
Can you also remove the `/doc/test.json` file? (check the diff)

**Comment by mehemken** [comment]
Ok I've made some changes

- removed the images
- merged the two plotting module sections

So this is all ready to go.

One thought though. I don't find it intuitive that the plotting module is nested under the style section. As a relatively new user this is confusing to me. Perhaps the plotting module warrants its own section? instead of:

```
Plotting
~~~~~~~~
```
Perhaps:
```
Plotting
--------
```

**Comment by jorisvandenbossche** [comment]
Regarding the header level, you are certainly right (I suppose that's an oversight). You can make that change.

**Comment by mehemken** [comment]
Ok thanks for the feedback Joris. This looks good to me but let me know if there's anything else I should do. I'm new to contributing to Pandas so if I'm leaving out any best practices or conventions I'd like to correct that.

**Review by jreback — APPROVED** [review]

**Pull request merged**